src/Microsoft.ML.ImageAnalytics/Imager.cs

```diff
@@ -0,0 +1,420 @@
+            }
+
+            _image = SKBitmap.Decode(imagePath);
+            if (_image is null)
+            {
+                throw new ArgumentException($"Invalid path", nameof(imagePath));
+            }
+
+            EnsureSupportedPixelFormat();
+        }
+
+        private Imager(SKBitmap image)
+        {
+            _image = image;
+            EnsureSupportedPixelFormat();
+        }
+
+        private void EnsureSupportedPixelFormat()
+        {
+            Debug.Assert(_image is not null);
+
+            // Most of the time SkiaSharp create images with Bgra8888 or Rgba8888 pixel format.
+            if (_image.Info.ColorType != SKColorType.Bgra8888 && _image.Info.ColorType != SKColorType.Rgba8888)
+            {
+                if (!_image.CanCopyTo(SKColorType.Bgra8888))
+                {
+                    throw new InvalidOperationException("Unsupported image format.");
+                }
+
+                SKBitmap image1 = _image.Copy(SKColorType.Bgra8888);
+                _image.Dispose();
+                _image = image1;
+            }
+        }
+
+        /// <summary>
+        /// Create BRGA32 pixel format imager object from the pixel data buffer span.
+        /// </summary>
+        /// <param name="width">The width of the image in pixels.</param>
+        /// <param name="height">The height of the image in pixels.</param>
+        /// <param name="imagePixelData">The pixels data to create the image from.</param>
+        /// <returns>Imager object.</returns>
+        public static unsafe Imager CreateFromBgra32PixelData(int width, int height, Span<byte> imagePixelData)
+        {
+            if (imagePixelData.Length != width * height * 4)
+            {
+                throw new ArgumentException($"Invalid {nameof(imagePixelData)} buffer size.");
+            }
+
+            SKBitmap image = new SKBitmap(new SKImageInfo(width, height, SKColorType.Bgra8888));
+
+            Debug.Assert(image.Info.BitsPerPixel == 32);
+            Debug.Assert(image.RowBytes * image.Height == width * height * 4);
+
+            imagePixelData.CopyTo(new Span<byte>(image.GetPixels().ToPointer(), image.Width * image.Height * 4));
+
+            return new Imager(image);
+        }
+
+        /// <summary>
+        /// Gets the image pixel data and how the colors are ordered in the used pixel format.
+        /// </summary>
+        /// <param name="alphaIndex">The index of the alpha in the pixel format.</param>
+        /// <param name="redIndex">The index of the red color in the pixel format.</param>
+        /// <param name="greenIndex">The index of the green color in the pixel format.</param>
+        /// <param name="blueIndex">The index of the blue color in the pixel format.</param>
+        /// <returns>The byte span containing the image pixel data.</returns>
+        public ReadOnlySpan<byte> Get32bbpImageData(out int alphaIndex, out int redIndex, out int greenIndex, out int blueIndex)
```

Comment:
This is a rather unusual shape for an API.   Is there a reason why we decided to expose a bunch of ints rather than just an enum value representing the format like SKIASharp did?   Seems to me like we could have a separate property to expose the image color type rather than force it here.

Reply:
The underlying ML.NET implementation always expects specific pixel formats to be able to operate on the data. I have ensured we always have such format when creating the image. So, with my implementation, we can have internally either `BGRA32` or `RGBA32` (which also are the defaults for SkiaSharp). I thought of exposing the pixel format as Enum but this is not really needed for ML.NET at all. If we think we'll use it in other scenarios, we can support it. My intention here is not to expose sophisticated imaging APIs to be used in generic scenario, I am just scoping that to the ML.NET while ensuring we can expand the APIs in the future if we need to without any constraints. I am open to change if you think it is better to apply your suggestion.

Reply:
What do you think about `ImageWrapper`? 

other ideas like `MLImage`, `ImageHolder`,  and `ImageObject`.

Reply:
This Get API that also returns a bunch of ints just felt awkward to me.  It’s combining two tasks - getting the buffer and the format.  Would it make sense to expose the format as a separate API since it can be down without any method call,  either an enum or Tuple property or 4 properties?

Reply:
Ok, I will think about it and fix it.

Reply:
done. I have introduced enum for a pixel format.